types/main.d.ts

```diff
@@ -0,0 +1,296 @@
+// I will gradually move the members up across this line.
+
+/** Declaration file generated by dts-gen */
+
+export class Constraint {
+    constructor(rhs: any, isUpperBound: any, index: any, model: any);
+
+    addTerm(coefficient: any, variable: any): any;
+
+    relax(weight: any, priority: any): void;
+
+    removeTerm(term: any): any;
+
+    setRightHandSide(newRhs: any): any;
+
+    setVariableCoefficient(newCoefficient: any, variable: any): any;
+}
+
+export class Model {
+    constructor(precision: any, name: any);
+
+    activateMIRCuts(useMIRCuts: any): void;
+
+    addVariable(cost: any, id: any, isInteger: any, isUnrestricted: any, priority: any): any;
+
+    debug(debugCheckForCycles: any): void;
+
+    equal(rhs: any): any;
+
+    getNumberOfIntegerVariables(): any;
+
+    greaterThan(rhs: any): any;
+
+    isFeasible(): any;
+
+    loadJson(jsonModel: any): any;
+
+    log(message: any): any;
+
+    maximize(): any;
+
+    minimize(): any;
+
+    removeConstraint(constraint: any): any;
+
+    removeVariable(variable: any): any;
+
+    restore(): any;
+
+    save(): any;
+
+    setCost(cost: any, variable: any): any;
+
+    smallerThan(rhs: any): any;
+
+    solve(): any;
+
+    updateConstraintCoefficient(constraint: any, variable: any, difference: any): any;
+
+    updateRightHandSide(constraint: any, difference: any): any;
+}
+
+export class Tableau {
+    constructor(precision: any);
+
+    addConstraint(constraint: any): void;
+
+    addCutConstraints(cutConstraints: any): void;
+
+    addVariable(variable: any): void;
+
+    applyCuts(branchingCuts: any): void;
+
+    applyMIRCuts(): void;
+
+    branchAndCut(): void;
+
+    checkForCycles(varIndexes: any): any;
+
+    computeFractionalVolume(ignoreIntegerValues: any): any;
+
+    copy(): any;
+
+    countIntegerValues(): any;
+
+    density(): any;
+
+    getFractionalVarWithLowestCost(): any;
+
+    getMostFractionalVar(): any;
+
+    getNewElementIndex(): any;
+
+    getSolution(): any;
+
+    initialize(width: any, height: any, variables: any, unrestrictedVars: any): void;
+
+    isIntegral(): any;
+
+    log(message: any, force: any): any;
+
+    phase1(): any;
+
+    phase2(): any;
+
+    pivot(pivotRowIndex: any, pivotColumnIndex: any): void;
+
+    removeConstraint(constraint: any): void;
+
+    removeVariable(variable: any): void;
+
+    restore(): void;
+
+    save(): void;
+
+    setEvaluation(): void;
+
+    setModel(model: any): any;
+
+    setOptionalObjective(priority: any, column: any, cost: any): any;
+
+    simplex(): any;
+
+    solve(): any;
+
+    updateConstraintCoefficient(constraint: any, variable: any, difference: any): void;
+
+    updateCost(variable: any, difference: any): void;
+
+    updateRightHandSide(constraint: any, difference: any): void;
+
+    updateVariableValues(): void;
+
+}
+
+export const External: {
+};
+
+export const Numeral: any;
+
+export const branchAndCut: {
+};
```

Comment:
These looks like exported modules from `main.js`. Are they **all** intended to be used by the outside world?

https://github.com/JWally/jsLPSolver/blob/18942b7b670f9aeacf12f81397ad16a4692c1b5c/src/main.js#L36-L43